core/domain/exp_domain.py

```diff
@@ -2964,6 +2964,49 @@ def delete_state(self, state_name):
 
         del self.states[state_name]
 
+    def get_trainable_states_dict(self, old_states):
+        """Retrieves the state names of all trainable states in an exploration
+        segregated into state names with changed and unchanged answer groups.
+
+        Args:
+            old_states: dict. Dictionary containing all State domain objects.
+
+        Returns:
+            dict. The trainable states dict. This dict has two keys representing
+                state names with changed and unchanged answer groups
+                respectively.
+        """
+        trainable_states_dict = {
+            'state_names_with_changed_answer_groups': [],
+            'state_names_with_unchanged_answer_groups': []
+        }
+        new_states = self.states
+        for l in new_states['Home'].interaction.answer_groups:
+            print l.outcome.feedback
+        for state_name in new_states:
+            new_state = new_states[state_name]
+            if not new_state.can_undergo_classification():
+                continue
+
+            # The case where a new state is added. When this happens, there wont
+            # be a corresponding state name in the older state dict.
+            if state_name not in old_states:
+                trainable_states_dict[
+                    'state_names_with_changed_answer_groups'].append(state_name)
+            old_state = old_states[state_name]
+            old_training_data = old_state.get_training_data()
+            new_training_data = new_state.get_training_data()
+            if new_training_data != old_training_data:
```

Comment:
I talked about this with @pranavsid98 and I think as of now it is fine to detect even a small change. Later we can modify this to detect more significant change. So ignore this comment.